### ISSUE TITLE
Optimize and add functionality to MQTTController

### ISSUE THREAD

**Issue by Soarnir** [DIT113-V22/group-04#50]
## Description

MQTTController needs to be updated to prevent connection issues, apparent solution at the time of writing is the utilization of the singleton pattern, Additionally, due to this new implementation, additional constraints will need to be added for text views. To prevent duplicate entries across multiple activity initializations when the application changes screens.

## Checklist

- [x]  The singleton pattern works perfectly with all classes
- [x]  No duplicate entries across multiple activity initializations.

[Part of Epic Feature 5](https://github.com/DIT113-V22/group-04/wiki/EF-5-Connectivity)

**Comment by Soarnir** [comment]
Additionally, a check will be added to prevent duplicate messages being sent over mqtt.

**Comment by Soarnir** [comment]
This can be considered feature complete, waiting on #53 to be merged to create new pr here.
